integrations-and-supported-tools/transformers/notebooks/Neptune_Transformers.ipynb

```diff
@@ -0,0 +1,389 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": [
+     "header",
+     "comment"
+    ]
+   },
+   "source": [
+    "# Using the 🤗 Transformers Integration\n",
+    "\n",
+    "Neptune provides an integration with 🤗 Transformers. So all you need to do to log metadata of the 🤗 Transformers training/finetuning is write a few lines of additional code and logging to Neptune is taken care of. \n",
+    "\n",
+    "You can integrate metadata tracking with Neptune either by:\n",
+    "* passing report_to = \"neptune\" to the Trainer arguments.\n",
+    "* setting up a Neptune callback and passing it to the Trainer callbacks.\n",
+    "\n",
+    "In this guide, we will look at the second option i.e. setting up a Neptune callback and passing it to the Trainer.\n",
+    "\n",
+    "By the end of this guide, you will be able to use the HuggingFace Transformers integration to log\n",
+    "* Train Loss\n",
+    "* Evaluation Loss\n",
+    "* Trainer parameters\n",
+    "* Model parameters\n",
+    "* Model checkpoint\n",
+    "\n",
+    "[See this example in Neptune](https://app.neptune.ai/o/showcase/org/project-text-summarization-hf/e/PROJ-138/dashboard/Custom-Dashboard-97370bc5-ee32-48ce-a630-fc806a370e13)\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Before you start\n",
+    "\n",
+    "This notebook example lets you try out Neptune anonymously, with zero setup.\n",
+    "\n",
+    "* If you're running the notebook on your local machine, you need to have [Python](https://www.python.org/downloads/) and [pip](https://pypi.org/project/pip/) installed.\n",
+    "* If you want to see the example logged to your own workspace instead:\n",
+    "    * Create a Neptune account → [Take me to registration](https://neptune.ai/register)\n",
+    "    * Create a Neptune project that you will use for tracking metadata → [Tell me more about projects](https://docs.neptune.ai/administration/projects)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Install Neptune and dependencies"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install neptune-client transformers==4.22.0 datasets==2.5.1 torch==1.11 scipy==1.7.2 scikit-learn==1.0.1 numpy==1.22.0"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": [
+     "comment"
+    ]
+   },
+   "source": [
+    "## Start a run\n",
+    "\n",
+    "To connect your script to Neptune and create a new run, we tell Neptune:\n",
+    "* **Who you are** - with a Neptune API token\n",
+    "* **Where to send your data** - to a Neptune project\n",
+    "\n",
+    "The cell below lets you record data to the public project [common/quickstarts](https://app.neptune.ai/common/quickstarts) as an anonymous user."
```

Comment:
Should we open the Transformers example project to anonymous logging and change to that? @SiddhantSadangi

Reply:
Nope. That project is used for our demos, and we don't want dirty runs there.

Reply:
Updated.